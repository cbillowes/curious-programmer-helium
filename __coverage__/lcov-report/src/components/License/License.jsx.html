<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/License/License.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/License</a> License.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">26x</span>
<span class="cline-any cline-yes">26x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">26x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from "react"
import ExternalLink from "../Link/ExternalLink"
import config from "../../../data/SiteConfig"
import "./License.scss"
&nbsp;
class License extends Component {
  render() {
    const dct = { "xmlns:dct": "http://purl.org/dc/terms/" }
    const cc = { "xmlns:cc": "http://creativecommons.org/ns#" }
&nbsp;
    return (
      &lt;span className="license"&gt;
        &lt;ExternalLink
          rel="license"
          to="http://creativecommons.org/licenses/by-sa/4.0/"
          className="license-icon"
        &gt;
          &lt;img
            alt="Creative Commons License"
            src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
            title="Creative Commons Attribution-ShareAlike 4.0 International License"
          /&gt;
        &lt;/ExternalLink&gt;
        &lt;p&gt;
          &lt;span {...dct} property="dct:title"&gt;
            {" "}
            {config.siteTitle} content{" "}
          &lt;/span&gt;{" "}
          by{" "}
          &lt;ExternalLink
            {...cc}
            to={config.siteUrl}
            property="cc:attributionName"
            rel="cc:attributionURL"
          &gt;
            {config.userName}
          &lt;/ExternalLink&gt;{" "}
          is licensed under a{" "}
          &lt;ExternalLink
            rel="license"
            to="http://creativecommons.org/licenses/by-sa/4.0/"
          &gt;
            Creative Commons Attribution-ShareAlike 4.0 International License
          &lt;/ExternalLink&gt;
          .
        &lt;/p&gt;
      &lt;/span&gt;
    )
  }
}
&nbsp;
export default License
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Jan 06 2019 14:17:11 GMT+0200 (South Africa Standard Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
