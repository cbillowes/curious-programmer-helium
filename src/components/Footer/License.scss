@import "../../layout/variables.scss";

.footer-container {
    .footer {
        .license-footer {
            font-size: 0.7em;
            font-style: italic;
            font-family: $font-family-serif;
            font-weight: 100;
            line-height: 1.75em;
        }

        p {
            margin: 0;
        }

        .baby {
            display: none;
        }

        // TODO: Get this 900px monstor into a variable,
        // better yet, the entire media query into a mixin
        @media screen and (max-width: 900px) {
            .baby {
                display: block;
            }

            .grand {
                display: none;
            }
        }
    }
}
