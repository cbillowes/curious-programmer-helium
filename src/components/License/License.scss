@import "../../layout/layout.scss";

.license {
    .license-icon {
        float: left;
        margin: 1em;
    }
}
