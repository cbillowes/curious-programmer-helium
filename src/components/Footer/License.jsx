import React, { Component } from "react"
import License from "../License/License"
import Link from "gatsby-link"
import "./License.scss"

class LicenseFooter extends Component {
  render() {
    return (
      <div className="license-footer">
        <License /> 
        {" "}
        <p>
          Also, there is a 
          {" "}
          <Link to="/license">License</Link>{" for the code "}
          and it's probably a good idea to read the
          {" "}
          <Link to="/privacy-policy">Privacy Policy</Link>
          .
        </p>
      </div>
    )
  }
}

export default LicenseFooter
