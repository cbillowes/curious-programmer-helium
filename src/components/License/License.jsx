import React, { Component } from "react"
import ExternalLink from "../Link/ExternalLink"
import config from "../../../data/SiteConfig"
import "./License.scss"

class License extends Component {
  render() {
    const dct = { "xmlns:dct": "http://purl.org/dc/terms/" }
    const cc = { "xmlns:cc": "http://creativecommons.org/ns#" }

    return (
      <span className="license">
        <ExternalLink
          rel="license"
          to="http://creativecommons.org/licenses/by-sa/4.0/"
          className="license-icon"
        >
          <img
            alt="Creative Commons License"
            src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
            title="Creative Commons Attribution-ShareAlike 4.0 International License"
          />
        </ExternalLink>
        <p>
          <span {...dct} property="dct:title">
            {" "}
            {config.siteTitle} content
            {" "}
          </span>
          {" "}
          by
          {" "}
          <ExternalLink
            {...cc}
            to={config.siteUrl}
            property="cc:attributionName"
            rel="cc:attributionURL"
          >
            {config.userName}
          </ExternalLink>
          {" "}
          is licensed under a
          {" "}
          <ExternalLink
            rel="license"
            to="http://creativecommons.org/licenses/by-sa/4.0/"
          >
            Creative Commons Attribution-ShareAlike 4.0 International License
          </ExternalLink>
          .
        </p>
      </span>
    )
  }
}

export default License
