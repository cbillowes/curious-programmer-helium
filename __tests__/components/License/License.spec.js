import React from "react"
import renderer from "react-test-renderer"
import { mount } from "enzyme"
import License from "../../../src/components/License/License"

describe("a license component", () => {
  let element = <License />
  let component
  
  beforeAll(() => {
    component = mount(element)    
  })

  it("should render", () => {
    const tree = renderer.create(element).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should render container", () => {
    expect(component.find(".license").length).toEqual(1)
  })

  it("should render creative commons license image", () => {
    expect(component.html()).toContain(`<img alt="Creative Commons License" src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png" title="Creative Commons Attribution-ShareAlike 4.0 International License">`)
  })

  it("should render terms property", () => {
    expect(component.html()).toContain(`<span xmlns:dct="http://purl.org/dc/terms/" property="dct:title"> Curious Programmer content </span>`)
  })

  it("should render attribution url", () => {
    expect(component.html()).toContain(`<a href="https://curiousprogrammer.io" class="" target="_blank" rel="nofollow noopener noreferrer cc:attributionURL">Clarice Bouwer</a>`)
  })

  it("should render licensed under url", () => {
    expect(component.html()).toContain(`<a href="http://creativecommons.org/licenses/by-sa/4.0/" class="" target="_blank" rel="nofollow noopener noreferrer license">Creative Commons Attribution-ShareAlike 4.0 International License</a>`)
  })
})