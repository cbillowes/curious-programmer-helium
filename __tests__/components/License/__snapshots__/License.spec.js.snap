// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a license component should render 1`] = `
<span
  className="license"
>
  <a
    className="license-icon"
    href="http://creativecommons.org/licenses/by-sa/4.0/"
    rel="nofollow noopener noreferrer license"
    target="_blank"
  >
    <img
      alt="Creative Commons License"
      src="https://i.creativecommons.org/l/by-sa/4.0/88x31.png"
      title="Creative Commons Attribution-ShareAlike 4.0 International License"
    />
  </a>
  <p>
    <span
      property="dct:title"
      xmlns:dct="http://purl.org/dc/terms/"
    >
       
      Curious Programmer
       content
       
    </span>
     
    by
     
    <a
      className=""
      href="https://curiousprogrammer.io"
      rel="nofollow noopener noreferrer cc:attributionURL"
      target="_blank"
    >
      Clarice Bouwer
    </a>
     
    is licensed under a
     
    <a
      className=""
      href="http://creativecommons.org/licenses/by-sa/4.0/"
      rel="nofollow noopener noreferrer license"
      target="_blank"
    >
      Creative Commons Attribution-ShareAlike 4.0 International License
    </a>
    .
  </p>
</span>
`;
